kroxylicious-sample/src/main/java/io/kroxylicious/sample/util/SampleFilterTransformer.java

```diff
@@ -84,7 +84,12 @@ private static ByteBuffer transformRecord(ByteBuffer in, String findValue, Strin
      * Instantiates a MemoryRecordsBuilder object using the given stream. This duplicates some of the
      * functionality in io.kroxylicious.proxy.internal, but we aren't supposed to import from there.
      */
-    private static MemoryRecordsBuilder createMemoryRecordsBuilder(ByteBufferOutputStream stream) {
+    private static MemoryRecordsBuilder createMemoryRecordsBuilder(ByteBufferOutputStream stream, RecordBatch firstBatch) {
+        if (firstBatch != null) {
```

Comment:
is `firstBatch == null` an error condition rather than something we should quietly accept?

Reply:
maybe the empty FetchResponse case when there are no new records

Reply:
We probably shouldn't get this deep with a null batch. So should maybe add a `@NotNull` [1] to firstBatch.



[1] Or equivalent from which ever annotation package we chose.

Reply:
I've opted to move the empty batch check further up, using `records.batchIterator().hasNext()` to see if we have any batches.